### PR TITLE
WIP: add pacemaker constraints needed for reboots

### PR DESCRIPTION
**Pull request by cwolferh**
This is still an early WIP.

An additional enhancement has been added where pacemaker IP resources are named like:

     ip-horizon-adm-192.168.201.95      (ocf::heartbeat:IPaddr2):       Started pcmk-c1a1
     ip-horizon-priv-192.168.201.94     (ocf::heartbeat:IPaddr2):       Started pcmk-c1a2
     ip-horizon-pub-192.168.7.100       (ocf::heartbeat:IPaddr2):       Started pcmk-c1a2

for sanity (e.g., instead of just ip-1.2.3.4).  I thought about removing the actual IP address from the resource name altogether (e.g. ip-horizon-pub) but A) I'm not sure this is really a huge win because it is nice seeing the IP in pcs status and B) a pretty large refactor to ha-all-in-one-util.bash would be needed.

The idea right now is to use quickstack::pacemaker::constraint::haproxy_vips as the way to define constraints between haproxy and the vips (pretty much every vip has a constraint on haproxy, so following the pattern we already use for vips makes sense here).  All other non-vip inter-service ordering constraints will live in quickstack::pacemaker::constraints.

### CODE REVIEW

File: puppet/modules/quickstack/manifests/pacemaker/constraint/typical.pp [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# just some sytantic sugar
+
+define quickstack::pacemaker::constraint::typical(
+  $first_resource  = '',
+  $second_resource = '',
+  $colocation      = true,
+  $anchor_me        = true
+) {
+  quickstack::pacemaker::constraint::base { $name :
```
Comment (cw4) [inline puppet/modules/quickstack/manifests/pacemaker/constraint/typical.pp]: Good to know, I will use $title instead.  Thanks!

Comment (cwolferh) [inline puppet/modules/quickstack/manifests/pacemaker/constraint/typical.pp]: Good to know, I will use $title instead. Thanks!

**Comment by cwolferh** [comment]
This is still a work in progress but I decided to move the constraints between haproxy and the vips into vips.pp itself (saves a lot of cutting and pasting).  The haproxy/vip constraint work is solid and takes care of all the haproxy/vip constraints but I still have more constraints to add to quickstack::pacemaker::constraints.

**Comment by cwolferh** [comment]
This is working, just have a little bit of clean up and commit squashing left.  Here are what the constraints look like with this PR:

    Location Constraints:
    Ordering Constraints:
      start haproxy-clone then start ip-ceilometer-pub-192.168.201.86
      start haproxy-clone then start ip-amqp-pub-192.168.201.13
      start haproxy-clone then start ip-ceilometer-adm-192.168.201.88
      start haproxy-clone then start ip-ceilometer-prv-192.168.201.87
      start haproxy-clone then start ip-horizon-adm-192.168.201.95
      start haproxy-clone then start ip-horizon-prv-192.168.201.94
      start haproxy-clone then start ip-galera-pub-192.168.201.7
      start haproxy-clone then start ip-horizon-pub-192.168.7.100
      start haproxy-clone then start ip-keystone-pub-192.168.201.33
      start haproxy-clone then start ip-keystone-prv-192.168.201.34
      start haproxy-clone then start ip-keystone-adm-192.168.201.35
      start haproxy-clone then start ip-glance-adm-192.168.201.25
      start haproxy-clone then start ip-glance-pub-192.168.201.23
      start haproxy-clone then start ip-glance-prv-192.168.201.24
      start fs-varlibglanceimages-clone then start openstack-glance-registry-clone
      start openstack-glance-registry-clone then start openstack-glance-api-clone
      start haproxy-clone then start ip-nova-pub-192.168.201.63
      start haproxy-clone then start ip-nova-prv-192.168.201.64
      start haproxy-clone then start ip-nova-adm-192.168.201.65
      start openstack-nova-consoleauth-clone then start openstack-nova-novncproxy-clone
      start openstack-nova-novncproxy-clone then start openstack-nova-api-clone
      start openstack-nova-api-clone then start openstack-nova-scheduler-clone
      start openstack-nova-scheduler-clone then start openstack-nova-conductor-clone
      start haproxy-clone then start ip-cinder-pub-192.168.201.83
      start haproxy-clone then start ip-cinder-adm-192.168.201.85
      start haproxy-clone then start ip-cinder-prv-192.168.201.84
      start openstack-cinder-scheduler-clone then start openstack-cinder-volume
      start openstack-cinder-api-clone then start openstack-cinder-scheduler-clone
      start haproxy-clone then start ip-heat-prv-192.168.201.114
      start haproxy-clone then start ip-heat-adm-192.168.201.115
      start haproxy-clone then start ip-heat_cfn-pub-192.168.201.123
      start haproxy-clone then start ip-heat_cfn-adm-192.168.201.125
      start haproxy-clone then start ip-heat-pub-192.168.201.113
      start haproxy-clone then start ip-heat_cfn-prv-192.168.201.124
      start openstack-heat-api-clone then start openstack-heat-api-cfn-clone
      start openstack-heat-api-cfn-clone then start openstack-heat-api-cloudwatch-clone
      start openstack-heat-api-cloudwatch-clone then start openstack-heat-engine
      start galera-master then start openstack-keystone-clone
      start rabbitmq-server-clone then start openstack-keystone-clone
      start haproxy-clone then start openstack-keystone-clone
      start openstack-keystone-clone then start openstack-nova-consoleauth-clone
      start memcached-clone then start openstack-keystone-clone
      start openstack-keystone-clone then start openstack-glance-registry-clone
      start openstack-keystone-clone then start openstack-cinder-api-clone
      start openstack-ceilometer-central then start openstack-ceilometer-collector-clone
      start openstack-ceilometer-collector-clone then start openstack-ceilometer-api-clone
      start openstack-ceilometer-api-clone then start ceilometer-delay-clone
      start ceilometer-delay-clone then start openstack-ceilometer-alarm-evaluator-clone
      start openstack-ceilometer-alarm-evaluator-clone then start openstack-ceilometer-alarm-notifier-clone
      start openstack-ceilometer-alarm-notifier-clone then start openstack-ceilometer-notification-clone
      start openstack-keystone-clone then start openstack-ceilometer-central
      start mongod-clone then start openstack-ceilometer-central
      start haproxy-clone then start ip-neutron-pub-192.168.201.103
      start haproxy-clone then start ip-neutron-adm-192.168.201.105
      start haproxy-clone then start ip-neutron-prv-192.168.201.104
      start neutron-ovs-cleanup-clone then start neutron-netns-cleanup-clone
      start neutron-netns-cleanup-clone then start neutron-agents
      start neutron-openvswitch-agent then start neutron-dhcp-agent
      start neutron-openvswitch-agent then start neutron-l3-agent
      start neutron-openvswitch-agent then start neutron-metadata-agent
      start openstack-keystone-clone then start neutron-server-clone
      start haproxy-clone then start ip-swift-pub-192.168.201.73
      start haproxy-clone then start ip-swift-prv-192.168.111.55
      start openstack-swift-proxy-clone then start openstack-swift-object-expirer
      start openstack-keystone-clone then start openstack-swift-proxy
    Colocation Constraints:
      ip-ceilometer-pub-192.168.201.86 with haproxy-clone
      ip-amqp-pub-192.168.201.13 with haproxy-clone
      ip-ceilometer-adm-192.168.201.88 with haproxy-clone
      ip-ceilometer-adm-192.168.201.88 with ip-ceilometer-pub-192.168.201.86
      ip-ceilometer-prv-192.168.201.87 with haproxy-clone
      ip-ceilometer-prv-192.168.201.87 with ip-ceilometer-pub-192.168.201.86
      ip-horizon-adm-192.168.201.95 with haproxy-clone
      ip-horizon-adm-192.168.201.95 with ip-horizon-pub-192.168.7.100
      ip-horizon-prv-192.168.201.94 with haproxy-clone
      ip-horizon-prv-192.168.201.94 with ip-horizon-pub-192.168.7.100
      ip-galera-pub-192.168.201.7 with haproxy-clone
      ip-horizon-pub-192.168.7.100 with haproxy-clone
      ip-keystone-adm-192.168.201.35 with haproxy-clone
      ip-keystone-adm-192.168.201.35 with ip-keystone-pub-192.168.201.33
      ip-keystone-pub-192.168.201.33 with haproxy-clone
      ip-keystone-prv-192.168.201.34 with haproxy-clone
      ip-keystone-prv-192.168.201.34 with ip-keystone-pub-192.168.201.33
      ip-glance-adm-192.168.201.25 with haproxy-clone
      ip-glance-pub-192.168.201.23 with haproxy-clone
      ip-glance-adm-192.168.201.25 with ip-glance-pub-192.168.201.23
      ip-glance-prv-192.168.201.24 with haproxy-clone
      ip-glance-prv-192.168.201.24 with ip-glance-pub-192.168.201.23
      openstack-glance-registry-clone with fs-varlibglanceimages-clone
      openstack-glance-api-clone with openstack-glance-registry-clone
      ip-nova-pub-192.168.201.63 with haproxy-clone
      ip-nova-prv-192.168.201.64 with haproxy-clone
      ip-nova-prv-192.168.201.64 with ip-nova-pub-192.168.201.63
      ip-nova-adm-192.168.201.65 with haproxy-clone
      ip-nova-adm-192.168.201.65 with ip-nova-pub-192.168.201.63
      openstack-nova-novncproxy-clone with openstack-nova-consoleauth-clone
      openstack-nova-api-clone with openstack-nova-novncproxy-clone
      openstack-nova-scheduler-clone with openstack-nova-api-clone
      openstack-nova-scheduler-clone with openstack-nova-conductor-clone
      ip-cinder-pub-192.168.201.83 with haproxy-clone
      ip-cinder-adm-192.168.201.85 with haproxy-clone
      ip-cinder-adm-192.168.201.85 with ip-cinder-pub-192.168.201.83
      ip-cinder-prv-192.168.201.84 with haproxy-clone
      ip-cinder-prv-192.168.201.84 with ip-cinder-pub-192.168.201.83
      openstack-cinder-volume with openstack-cinder-scheduler-clone
      openstack-cinder-scheduler-clone with openstack-cinder-api-clone
      ip-heat-prv-192.168.201.114 with haproxy-clone
      ip-heat-prv-192.168.201.114 with ip-heat-pub-192.168.201.113
      ip-heat-adm-192.168.201.115 with haproxy-clone
      ip-heat-adm-192.168.201.115 with ip-heat-pub-192.168.201.113
      ip-heat_cfn-adm-192.168.201.125 with haproxy-clone
      ip-heat_cfn-adm-192.168.201.125 with ip-heat_cfn-pub-192.168.201.123
      ip-heat_cfn-pub-192.168.201.123 with haproxy-clone
      ip-heat-pub-192.168.201.113 with haproxy-clone
      ip-heat_cfn-prv-192.168.201.124 with haproxy-clone
      ip-heat_cfn-prv-192.168.201.124 with ip-heat_cfn-pub-192.168.201.123
      openstack-heat-api-cfn-clone with openstack-heat-api-clone
      openstack-heat-api-cloudwatch-clone with openstack-heat-api-cfn-clone
      openstack-heat-engine with openstack-heat-api-cloudwatch-clone
      ip-neutron-pub-192.168.201.103 with haproxy-clone
      ip-neutron-adm-192.168.201.105 with haproxy-clone
      ip-neutron-adm-192.168.201.105 with ip-neutron-pub-192.168.201.103
      ip-neutron-prv-192.168.201.104 with haproxy-clone
      ip-neutron-prv-192.168.201.104 with ip-neutron-pub-192.168.201.103
      neutron-dhcp-agent with neutron-openvswitch-agent
      neutron-l3-agent with neutron-openvswitch-agent
      neutron-metadata-agent with neutron-openvswitch-agent
      ip-swift-prv-192.168.111.55 with haproxy-clone
      ip-swift-prv-192.168.111.55 with ip-swift-pub-192.168.201.73
      ip-swift-pub-192.168.201.73 with haproxy-clone

**Comment by cwolferh** [comment]
Closing in favor of https://github.com/redhat-openstack/astapor/pull/452

**Pull request closed without merging**